core/src/main/java/com/amplifyframework/devmenu/DevMenuLogsFragment.java

```diff
@@ -54,11 +73,55 @@ public boolean onQueryTextSubmit(String searchText) {
 
             @Override
             public boolean onQueryTextChange(String searchText) {
-                logsText.setText(R.string.placeholder_logs);
-                logsText.setText(developerMenu.getFilteredLogs(searchText));
+                searchQuery = searchText;
+                displayFilteredLogs();
                 return true;
             }
         });
+        logLevelMenuId = R.id.all_logs;
+        Button filterLogs = logsView.findViewById(R.id.filter_logs);
+        registerForContextMenu(filterLogs);
+        filterLogs.setOnClickListener(view -> requireActivity().openContextMenu(view));
         return logsView;
     }
+
+    @Override
+    public void onCreateContextMenu(@NonNull ContextMenu menu, @NonNull View view,
+                                    @Nullable ContextMenu.ContextMenuInfo menuInfo) {
+        super.onCreateContextMenu(menu, view, menuInfo);
+        requireActivity().getMenuInflater().inflate(R.menu.dev_menu_logs_menu, menu);
+        menu.findItem(logLevelMenuId).setChecked(true);
+    }
+
+    @Override
+    public boolean onContextItemSelected(MenuItem item) {
+        int itemId = item.getItemId();
+        if (itemId == R.id.all_logs) {
+            selectedLogLevel = null;
+        } else if (itemId == R.id.error_logs) {
+            selectedLogLevel = LogLevel.ERROR;
+        } else if (itemId == R.id.warn_logs) {
+            selectedLogLevel = LogLevel.WARN;
+        } else if (itemId == R.id.info_logs) {
+            selectedLogLevel = LogLevel.INFO;
+        } else if (itemId == R.id.debug_logs) {
+            selectedLogLevel = LogLevel.DEBUG;
+        } else if (itemId == R.id.verbose_logs) {
+            selectedLogLevel = LogLevel.VERBOSE;
+        } else {
+            return super.onContextItemSelected(item);
+        }
+        logLevelMenuId = itemId;
+        displayFilteredLogs();
+        return true;
+    }
+
+    /**
+     * Display the logs that contain the provided search query and
+     * whose log level matches the selected log level.
+     */
+    private void displayFilteredLogs() {
+        logsText.setText(R.string.placeholder_logs);
+        logsText.setText(developerMenu.getFilteredLogs(searchQuery, selectedLogLevel));
```

Comment:
What's the purpose for setting the placeholder text first before setting it to getFilteredLogs?

Reply:
In a previous PR Jameson commented that it would be helpful to display some placeholder text (i.e. "Search results will appear here...") in case it takes time to display the logs (https://github.com/aws-amplify/amplify-android/pull/659#discussion_r459085658), although that comment was more related to when you initially navigate to the logs view. If it does take time to display the logs, I thought it could be helpful to display the placeholder text indicating that search results are being retrieved, otherwise it may seem like the search functionality is broken.